relay_handler.js

```diff
@@ -105,16 +109,15 @@ RelayHandler.prototype.handleRequest = function handleRequest(req, buildRes) {
     // }
 
     if (self.circuits) {
-        var result = self.circuits.getCircuitForRequest(req);
-        if (result.err) {
-            var errFrame = result.err;
-            buildRes().sendError(errFrame.codeName, errFrame.message);
+        var circuit = self.circuits.getCircuit(req.headers.cn || 'no-cn', req.serviceName, String(req.arg1));
+        if (!circuit.state.shouldRequest()) {
```

Comment:
`getCircuitForRequest` would also send an error frame if it couldn't find a serviceName, but ServiceDispatchHandler does the same check earlier, so it was unnecessary.